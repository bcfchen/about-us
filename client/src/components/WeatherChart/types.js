import PropTypes from "prop-types";
import HourlyForecast from "../../models/HourlyForecast";

export const propTypes = {
    isLoading: PropTypes.bool.isRequired,
    hourlyForecasts: PropTypes.arrayOf(PropTypes.instanceOf(HourlyForecast)).isRequired
};
