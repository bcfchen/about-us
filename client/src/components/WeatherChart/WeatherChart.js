import React from "react";
import { render } from "react-dom";
import Highcharts from "highcharts/highstock";
import HighchartsReact from "highcharts-react-official";

const WeatherChart = ({ hourlyForecasts }) => {
    const options = {
        title: {
            text: null,
        },
        xAxis: {
            visible: false
        },
        yAxis: {
            visible: false
        },
        plotOptions: {
            series: {
                showInLegend: false,
                marker: {
                    enabled: false
                }
            }
        },
        series: [{
            data: hourlyForecasts.map(forecast => forecast.temperature),
            type: "spline"
        }]
    }

    return (<div>
        <HighchartsReact
            highcharts={Highcharts}
            options={options}
        />
    </div>);
}

export default WeatherChart;