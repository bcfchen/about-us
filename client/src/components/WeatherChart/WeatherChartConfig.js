export default {
    responsive: {
        rules: [{
            condition: {
                maxHeight: 1
            }
        }]
    },
    chart: {
        backgroundColor: null,
        height: "180px"
    },
    title: {
        text: null,
    },
    tooltip: {
        enabled: false
    },
    xAxis: {
        visible: false
    },
    yAxis: {
        visible: false
    },
    plotOptions: {
        series: {
            showInLegend: false,
            marker: {
                enabled: false
            }
        }
    }
};