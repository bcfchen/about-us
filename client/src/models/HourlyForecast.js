import moment from "moment";

class HourlyForecast {
    constructor(properties, location) {
        this.temperature = properties.Temperature.Value;
        this.text = properties.IconPhrase;
        this.dateTime = properties.DateTime;
        this.windSpeed = properties.Wind.Speed.Value + " km/h";
        this.humidity = properties.RelativeHumidity + " %";
        this.uvIndex = properties.UVIndex;
        this.location = location;
    }

    getHoursAndMinutes() {
        return moment(this.dateTime).format("HH:mm A");
    }

    getLocationString() {
        return this.location.getLocationString();
    }
}

export default HourlyForecast;