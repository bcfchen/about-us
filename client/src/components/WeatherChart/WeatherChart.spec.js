import React from "react";
import { shallow } from "enzyme";
import WeatherChart from "./WeatherChart";
import Highcharts from "highcharts/highstock";
import HourlyForecast from "../../models/HourlyForecast";
jest.mock("highcharts/highstock", () => () => { });

describe("<WeatherChart/>", () => {
    let defaultWeatherProps = {
        Temperature: {
            Value: 15
        },
        Wind: {
            Speed: {
                Value: 6
            }
        },
        IconPhrase: "Sunny"
    };
    let defaultProps = {
        isLoading: true,
        hourlyForecasts: [new HourlyForecast(defaultWeatherProps)]
    };

    let component = null;
    let render = (props = defaultProps) => {
        component = shallow(<WeatherChart {...props} />);
    };

    describe("on page load", () => {
        it("should render correctly", () => {
            render();
            expect(component).toMatchSnapshot();
        });
    });
});
