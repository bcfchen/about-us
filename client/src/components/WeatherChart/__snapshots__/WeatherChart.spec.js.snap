// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherChart/> on page load should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WeatherChart
    hourlyForecasts={
        Array [
            HourlyForecast {
              "dateTime": undefined,
              "humidity": "undefined %",
              "isDaylight": undefined,
              "location": undefined,
              "temperature": 15,
              "text": "Sunny",
              "uvIndex": undefined,
              "windSpeed": "6 km/h",
            },
          ]
    }
    isLoading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="weather-chart"
>
        <Unknown
                highcharts={[Function]}
                options={
                        Object {
                                "chart": Object {
                                  "backgroundColor": null,
                                  "height": "180px",
                                },
                                "plotOptions": Object {
                                  "series": Object {
                                    "marker": Object {
                                      "enabled": false,
                                    },
                                    "showInLegend": false,
                                  },
                                },
                                "responsive": Object {
                                  "rules": Array [
                                    Object {
                                      "condition": Object {
                                        "maxHeight": 1,
                                      },
                                    },
                                  ],
                                },
                                "series": Array [
                                  Object {
                                    "color": "#fff",
                                    "data": Array [
                                      15,
                                    ],
                                    "type": "spline",
                                  },
                                ],
                                "title": Object {
                                  "text": null,
                                },
                                "tooltip": Object {
                                  "enabled": false,
                                },
                                "xAxis": Object {
                                  "visible": false,
                                },
                                "yAxis": Object {
                                  "visible": false,
                                },
                              }
                }
        />
</div>,
      "in": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Unknown
          highcharts={[Function]}
          options={
                    Object {
                              "chart": Object {
                                "backgroundColor": null,
                                "height": "180px",
                              },
                              "plotOptions": Object {
                                "series": Object {
                                  "marker": Object {
                                    "enabled": false,
                                  },
                                  "showInLegend": false,
                                },
                              },
                              "responsive": Object {
                                "rules": Array [
                                  Object {
                                    "condition": Object {
                                      "maxHeight": 1,
                                    },
                                  },
                                ],
                              },
                              "series": Array [
                                Object {
                                  "color": "#fff",
                                  "data": Array [
                                    15,
                                  ],
                                  "type": "spline",
                                },
                              ],
                              "title": Object {
                                "text": null,
                              },
                              "tooltip": Object {
                                "enabled": false,
                              },
                              "xAxis": Object {
                                "visible": false,
                              },
                              "yAxis": Object {
                                "visible": false,
                              },
                            }
          }
/>,
        "className": "weather-chart",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "highcharts": [Function],
          "options": Object {
            "chart": Object {
              "backgroundColor": null,
              "height": "180px",
            },
            "plotOptions": Object {
              "series": Object {
                "marker": Object {
                  "enabled": false,
                },
                "showInLegend": false,
              },
            },
            "responsive": Object {
              "rules": Array [
                Object {
                  "condition": Object {
                    "maxHeight": 1,
                  },
                },
              ],
            },
            "series": Array [
              Object {
                "color": "#fff",
                "data": Array [
                  15,
                ],
                "type": "spline",
              },
            ],
            "title": Object {
              "text": null,
            },
            "tooltip": Object {
              "enabled": false,
            },
            "xAxis": Object {
              "visible": false,
            },
            "yAxis": Object {
              "visible": false,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="weather-chart"
>
          <Unknown
                    highcharts={[Function]}
                    options={
                              Object {
                                        "chart": Object {
                                          "backgroundColor": null,
                                          "height": "180px",
                                        },
                                        "plotOptions": Object {
                                          "series": Object {
                                            "marker": Object {
                                              "enabled": false,
                                            },
                                            "showInLegend": false,
                                          },
                                        },
                                        "responsive": Object {
                                          "rules": Array [
                                            Object {
                                              "condition": Object {
                                                "maxHeight": 1,
                                              },
                                            },
                                          ],
                                        },
                                        "series": Array [
                                          Object {
                                            "color": "#fff",
                                            "data": Array [
                                              15,
                                            ],
                                            "type": "spline",
                                          },
                                        ],
                                        "title": Object {
                                          "text": null,
                                        },
                                        "tooltip": Object {
                                          "enabled": false,
                                        },
                                        "xAxis": Object {
                                          "visible": false,
                                        },
                                        "yAxis": Object {
                                          "visible": false,
                                        },
                                      }
                    }
          />
</div>,
        "in": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            highcharts={[Function]}
            options={
                        Object {
                                    "chart": Object {
                                      "backgroundColor": null,
                                      "height": "180px",
                                    },
                                    "plotOptions": Object {
                                      "series": Object {
                                        "marker": Object {
                                          "enabled": false,
                                        },
                                        "showInLegend": false,
                                      },
                                    },
                                    "responsive": Object {
                                      "rules": Array [
                                        Object {
                                          "condition": Object {
                                            "maxHeight": 1,
                                          },
                                        },
                                      ],
                                    },
                                    "series": Array [
                                      Object {
                                        "color": "#fff",
                                        "data": Array [
                                          15,
                                        ],
                                        "type": "spline",
                                      },
                                    ],
                                    "title": Object {
                                      "text": null,
                                    },
                                    "tooltip": Object {
                                      "enabled": false,
                                    },
                                    "xAxis": Object {
                                      "visible": false,
                                    },
                                    "yAxis": Object {
                                      "visible": false,
                                    },
                                  }
            }
/>,
          "className": "weather-chart",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "highcharts": [Function],
            "options": Object {
              "chart": Object {
                "backgroundColor": null,
                "height": "180px",
              },
              "plotOptions": Object {
                "series": Object {
                  "marker": Object {
                    "enabled": false,
                  },
                  "showInLegend": false,
                },
              },
              "responsive": Object {
                "rules": Array [
                  Object {
                    "condition": Object {
                      "maxHeight": 1,
                    },
                  },
                ],
              },
              "series": Array [
                Object {
                  "color": "#fff",
                  "data": Array [
                    15,
                  ],
                  "type": "spline",
                },
              ],
              "title": Object {
                "text": null,
              },
              "tooltip": Object {
                "enabled": false,
              },
              "xAxis": Object {
                "visible": false,
              },
              "yAxis": Object {
                "visible": false,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
